src/modules/job-info/job_state.c

```diff
@@ -1745,45 +1677,126 @@ static int job_events_annotations (struct job_state_ctx *jsctx,
+                return -1;
+        }
+        else if (!strcmp (name, "alloc")) {
+            if (job_events_alloc (jsctx,
+                                  id,
+                                  eventlog_seq,
+                                  timestamp) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "finish")) {
+            if (job_events_finish (jsctx,
+                                   id,
+                                   eventlog_seq,
+                                   timestamp,
+                                   context) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "clean")) {
+            if (job_events_clean (jsctx,
+                                  id,
+                                  eventlog_seq,
+                                  timestamp) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "priority")) {
+            if (job_events_priority (jsctx,
+                                     id,
+                                     eventlog_seq,
+                                     timestamp,
+                                     context) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "exception")) {
+            if (job_events_exception (jsctx,
+                                      id,
+                                      eventlog_seq,
+                                      timestamp,
+                                      context) < 0)
+                return -1;
         }
         else if (!strcmp (name, "annotations")) {
             if (job_events_annotations (jsctx, id, context) < 0)
-                goto error;
+                return -1;
         }
+        else {
+            struct job *job;
+            if (!(job = zhashx_lookup (jsctx->index, &id))) {
+                flux_log_error (jsctx->h, "%s: job %ju not in hash",
+                                __FUNCTION__, (uintmax_t)id);
+                return -1;
+            }
+            (void) job_events_update_eventlog_seq (jsctx, job, eventlog_seq);
+        }
+    }
+
+    return 0;
+}
+
+static void job_events_continuation (flux_future_t *f, void *arg)
+{
+    struct job_state_ctx *jsctx = arg;
+    json_t *events;
+
+    if (flux_rpc_get_unpack (f, "{s:o}", "events", &events) < 0) {
+        flux_log_error (jsctx->h, "%s: flux_rpc_get_unpack", __FUNCTION__);
+        goto error;
+    }
+
+    if (!json_is_array (events)) {
+        flux_log (jsctx->h, LOG_ERR, "%s: events EPROTO", __FUNCTION__);
+        goto error;
+    }
+
+    if (jsctx->pause) {
+        json_t *o = json_incref (events);
+        if (!zlistx_add_end (jsctx->events_journal_backlog, o)) {
+            flux_log_error (jsctx->h, "%s: zlistx_add_end", __FUNCTION__);
+            json_decref (o);
+            goto error;
+        }
+    }
+    else {
+        if (job_events_process (jsctx, events) < 0)
+            goto error;
```

Comment:
errno is not set for the `!json_is_array()` case nor the `zlistx_add_end()` failure cases.  The other cases need an audit.

Errno needs to be set before calliing `flux_reactor_stop_error()` so that `flux_reactor_run()`  has that errno set when it returns -1.